crates/proto/src/rr/domain/try_parse_ip.rs

```diff
@@ -18,24 +18,25 @@ pub trait TryParseIp {
 
 impl TryParseIp for str {
     fn try_parse_ip(&self) -> Option<RData> {
-        match IpAddr::from_str(self) {
-            Ok(IpAddr::V4(ip4)) => Some(RData::A(ip4)),
-            Ok(IpAddr::V6(ip6)) => Some(RData::AAAA(ip6)),
-            Err(_) => None,
-        }
+        RData::from_str(self).ok()
     }
 }
 
-// impl<'a> TryParseIp for &'a str {
-//     fn try_parse_ip(&self) -> Option<RData> {
-//         TryParseIp::try_parse_ip(*self)
-//     }
-// }
+impl FromStr for RData {
```

Comment:
Hm, I don't know if this one exactly makes sense, and it in some ways conflicts with other impls that handle all RDATA.

In this context, we are looking specifically to see if the address is already an IP such that resolution can be skipped. Semantically, I think that means that it should not be a blanket impl for RData when it only handles these two cases. Does that make sense?

Reply:
Yep, fair enough.